core/runtime/pom.xml

```diff
@@ -150,6 +150,19 @@
                         <parentFirstArtifact>org.junit.platform:junit-platform-commons</parentFirstArtifact>
                         <parentFirstArtifact>org.junit.platform:junit-platform-engine</parentFirstArtifact>
                     </parentFirstArtifacts>
+                    <runnerParentFirstArtifacts>
+                        <runnerParentFirstArtifact>io.quarkus:quarkus-bootstrap-runner</runnerParentFirstArtifact>
+                        <runnerParentFirstArtifact>io.quarkus:quarkus-development-mode-spi</runnerParentFirstArtifact>
+                        <!-- logging dependencies always need to be loaded by the JDK ClassLoader -->
+                        <runnerParentFirstArtifact>org.jboss.logging:jboss-logging</runnerParentFirstArtifact>
+                        <runnerParentFirstArtifact>org.jboss.logmanager:jboss-logmanager-embedded</runnerParentFirstArtifact>
+                        <runnerParentFirstArtifact>org.jboss.slf4j:slf4j-jboss-logmanager</runnerParentFirstArtifact>
+                        <runnerParentFirstArtifact>org.slf4j:slf4j-api</runnerParentFirstArtifact>
+                        <!-- The following dependencies are dependencies of the JBoss Logging so they too need  be loaded by the JDK ClassLoader -->
+                        <runnerParentFirstArtifact>io.smallrye.common:smallrye-common-io</runnerParentFirstArtifact>
+                        <runnerParentFirstArtifact>org.wildfly.common:wildfly-common</runnerParentFirstArtifact>
+                        <runnerParentFirstArtifact>org.graalvm.sdk:graal-sdk</runnerParentFirstArtifact>
+                    </runnerParentFirstArtifacts>
```

Comment:
Are you sure about all of these?  In particular the SLF4J ones shouldn't be necessary, nor should the SmallRye one.  Also the `svm` one shouldn't be needed either since it's only used for annotations AFAICT.

Reply:
Right, the slf4j stuff isn't necessary.

The smallrye stuff is necessary however (see the updated comments)

Reply:
What about the graalvm SDK?  What needs that at run time?

Reply:
That one is needed because of: https://github.com/quarkusio/quarkus/blob/966b5045e8f874060be06c6c7aaa0a074a93bd9b/independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/logging/InitialConfigurator.java#L51

Reply:
Ah I see.  I wonder if that's worth replacing it with the contents of that method:

```java
        if (System.getProperty("org.graalvm.nativeimage.imagecode").equals("runtime")) {
```

Reply:
Good idea! Let me do that

Reply:
I pushed a commit that removes the dependency on the GraalVM jar

Reply:
Hmmm. What does it bring to avoid having it there?

I'm not sure it's such a good idea to not rely on proper APIs.

Reply:
Just minimizing the class path size, particularly with a JAR that is never really used.  Note that the system property name and value are both stable APIs as well.